### PR TITLE
Automatic infinite gas abstraction

### PR DESCRIPTION
**Pull request by nwatson22**
This adds the option `--auto-abstract-gas` to `kevm prove` and `kevm foundry-prove`. When enabled, this option converts the term inside `#gas` to a variable for every new node (and does the same with the `<refund>` cell) to improve readability and/or performance.

### CODE REVIEW

File: kevm-pyk/pyproject.toml [hunk 1]
```diff
@@ -13,7 +13,7 @@ authors = [
 [tool.poetry.dependencies]
 python = "^3.10"
 pathos = "*"
-pyk = { git = "https://github.com/runtimeverification/pyk.git", tag="v0.1.330" }
+pyk = { git = "https://github.com/runtimeverification/pyk.git", branch = "noah/node-postprocessing" }
```
Comment (anvacaru) [inline kevm-pyk/pyproject.toml:16]: typo?

File: kevm-pyk/src/kevm_pyk/__main__.py [hunk 2]
```diff
@@ -495,6 +496,7 @@ def exec_foundry_prove(
     smt_retry_limit: int | None = None,
     failure_info: bool = True,
     trace_rewrites: bool = False,
+    auto_abstract_gas: bool = False,
     **kwargs: Any,
 ) -> None:
```
Comment (ehildenb) [inline kevm-pyk/src/kevm_pyk/__main__.py:501]: We should also expose this option via `kevm prove ...` too (in `exec_prove`).

File: kevm-pyk/src/kevm_pyk/utils.py [hunk 3]
```diff
@@ -176,6 +185,22 @@ def print_failure_info(proof: APRProof, kcfg_explore: KCFGExplore) -> list[str]:
...
+                result = KApply('<gas>', KApply('infGas', abstract_term_safely(term)))
+                return result
+            return term
+        elif type(term) is KApply and term.label.name == '<refund>':
+            return KApply('<refund>', KVariable('ABSTRACTED_REFUND', KSort('Int')))
+        else:
+            return term
+
+    return CTerm(config=bottom_up(_replace, cterm.config), constraints=cterm.constraints)
+
```
[the remark this answers is not in the document]
Comment (ehildenb) [reply kevm-pyk/src/kevm_pyk/utils.py:202]: Let's instead implement `CTerm.set_cell` in pyk, and then use that here for setting the `<gas>` and `<refund>` cells.

[the remark this answers is not in the document]
Comment (ehildenb) [reply kevm-pyk/src/kevm_pyk/utils.py:202]: You could also have `set_cell` trim any constraints that become irrelevant over the new cell.

[the remark this answers is not in the document]
Comment (ehildenb) [reply kevm-pyk/src/kevm_pyk/utils.py:202]: This probably looks fine for now, can refactor it later.

**Review by ehildenb — APPROVED** [review]

**Pull request merged**